packages/viewer/src/components/viewer-transform-widget/viewer-transform-widget.tsx

```diff
@@ -19,6 +28,12 @@ import { TransformWidget } from './widget';
   shadow: true,
 })
 export class ViewerTransformWidget {
+  /**
+   * An event that is emitted when the position of the widget changes.
+   */
+  @Event({ bubbles: true })
+  public positionChanged!: EventEmitter<Vector3.Vector3>;
```

Comment:
Should this be:
```typescript
public positionChanged!: EventEmitter<Vector3.Vector3 | undefined>;
```
? Or potentially just in an `else` on the condition above where we emit it if it'll never change to `undefined` internally. I'm actually a bit surprised that the `emit` doesn't complain that it could be undefined

Reply:
Yeah, I had a similar thought. By default, the emitter accepts optional values:
```
export interface EventEmitter<T = any> {
    emit: (data?: T) => CustomEvent<T>;
}
```

I think preventing an emit on an undefined value is a fine approach here.